src/test/scala/com/exasol/common/AbstractPropertiesTest.scala

```diff
@@ -166,37 +164,53 @@ class AbstractPropertiesTest extends AnyFunSuite with BeforeAndAfterEach with Mo
     val thrown = intercept[IllegalArgumentException] {
       BaseProperties(properties).getConnectionInformation(None)
     }
-    assert(thrown.getMessage === "Exasol metadata is None!")
+    assert(thrown.getMessage() === "Exasol metadata is None!")
   }
 
   test("getConnectionInformation returns storage connection information") {
     properties = Map("CONNECTION_NAME" -> "connection_info")
     val metadata = mock[ExaMetadata]
     val connectionInfo = newConnectionInformation("user", "secret")
     when(metadata.getConnection("connection_info")).thenReturn(connectionInfo)
-    assert(
-      BaseProperties(properties)
-        .getConnectionInformation(Option(metadata)) === connectionInfo
-    )
+    assert(BaseProperties(properties).getConnectionInformation(Option(metadata)) === connectionInfo)
     verify(metadata, times(1)).getConnection("connection_info")
   }
 
   test("parseConnectionInfo returns key value pairs from password") {
     properties = Map("CONNECTION_NAME" -> "connection_info")
-    val metadata = mockMetadata("", "a=secret1;b=secret2")
-    val result = BaseProperties(properties)
-      .parseConnectionInfo("username", Option(metadata))
-    assert(result === Map("a" -> "secret1", "b" -> "secret2"))
-    verify(metadata, times(1)).getConnection("connection_info")
+    val expected = Map("a" -> "secret1", "b" -> "secret2")
+    assertParseConnectionInfo("", "a=secret1;b=secret2", properties, expected)
   }
 
   test("parseConnectionInfo returns key value pairs with updated username") {
     properties = Map("CONNECTION_NAME" -> "connection_info")
-    val metadata = mockMetadata("John", "a=secret1;b=secret2")
-    val result = BaseProperties(properties)
-      .parseConnectionInfo("username", Option(metadata))
-    assert(result === Map("username" -> "John", "a" -> "secret1", "b" -> "secret2"))
+    val expected = Map("userKey" -> "John", "a" -> "secret1", "b" -> "secret2")
+    assertParseConnectionInfo("John", "a=secret1;b=secret2", properties, expected)
+  }
+
+  test("parseConnectionInfo returns key value pairs with custom separators") {
+    properties = Map("CONNECTION_NAME" -> "connection_info", "CONNECTION_SEPARATOR" -> "#")
+    val expected = Map("k1" -> "v1", "k2" -> "v2", "k3" -> "v3;k4=v4")
+    assertParseConnectionInfo("", "k1=v1#k2=v2#k3=v3;k4=v4", properties, expected)
+  }
+
+  test("parseConnectionInfo returns key value pairs with custom key-value assignment") {
+    properties = Map("CONNECTION_NAME" -> "connection_info", "CONNECTION_KEYVALUE_ASSIGNMENT" -> "@@")
+    val expected = Map("k1" -> "v1", "k2" -> "v2", "k3" -> "v3", "k4" -> "v4")
+    assertParseConnectionInfo("", "k1@@v1;k2@@v2;k3@@v3;k4@@v4", properties, expected)
+  }
```

Comment:
Why do we need this feature?

Reply:
Maybe you can add an explanation to a design file?
My first impression is: Keep the API simple. Don't add too many convenience features, that make it confusing. If customers want't different format, they can convert...

Reply:
I see.
Maybe you can document it somewhere? (like in a design.md)

Reply:
Yes, since this only required for Event Hubs, I am going to include it in https://github.com/exasol/kafka-connector-extension/issues/55